src/schemas/common.ts

```diff
@@ -0,0 +1,134 @@
+import { z } from "zod";
+import * as constants from "../constants";
+import {
+  AppendExpectedRevision,
+  BinaryEventType,
+  Credentials,
+  EventType,
+  ExpectedRevision,
+  Filter,
+  FilterBase,
+  FilterOn,
+  JSONEventType,
+  JSONType,
+  MetadataType,
+  Position,
+  ReadPosition,
+  ReadRevision,
+  RegexFilter,
+} from "../types";
+
+export const credentials = z.object({
+  username: z.string(),
+  password: z.string(),
+}) satisfies z.ZodType<Credentials>;
+
+export const baseOptions = z.object({
+  credentials: credentials.optional(),
+  requiresLeader: z.boolean().optional(),
+  deadline: z.number().min(0).optional(),
+}) satisfies z.ZodType;
+
+export const jsonType = z.union([
+  z.record(z.union([z.string(), z.number()]), z.unknown()),
+  z.array(z.unknown()),
+  z.string(),
+]) satisfies z.ZodType<JSONType>;
+
+export const metadataType = z.union([
+  jsonType,
+  z.instanceof(Uint8Array),
+]) satisfies z.ZodType<MetadataType>;
+
+export const jsonEventType = z.object({
+  type: z.string(),
+  data: jsonType,
+  metadata: metadataType.optional(),
+}) satisfies z.ZodType<JSONEventType>;
+
+export const binaryEventType = z.object({
+  type: z.string(),
+  data: z.instanceof(Uint8Array),
+  metadata: metadataType.optional(),
+}) satisfies z.ZodType<BinaryEventType>;
+
+export const eventType = z.union([
+  jsonEventType,
+  binaryEventType,
+]) satisfies z.ZodType<EventType>;
+
+export const jsonEventData = z.object({
+  id: z.string().uuid(),
+  contentType: z.literal("application/json"),
+  type: z.string(),
+  data: jsonType,
```

Comment:
```suggestion
  data: jsonType,
```

I am confused about this type 🤔 What is the motivation on enforcing a given `data` payload structure from the perspective of the ESDB Client?

Reply:
It should be any.